sources/packages/web/src/views/aest/institution/DesignationAESTView.vue

```diff
@@ -0,0 +1,221 @@
+
+    const loadDesignation = async () => {
+      designationAgreement.value = await DesignationAgreementService.shared.getDesignationAgreement(
+        props.designationAgreementId,
+      );
+
+      designationFormModel.institutionName =
+        designationAgreement.value.institutionName;
+      designationFormModel.institutionType =
+        designationAgreement.value.institutionType;
+      designationFormModel.isBCPrivate = designationAgreement.value.isBCPrivate;
+      designationFormModel.viewMode = DesignationFormViewModes.viewOnly;
+      designationFormModel.designationStatus =
+        designationAgreement.value.designationStatus;
+      designationFormModel.designationStatusClass = mapDesignationChipStatus(
+        designationAgreement.value.designationStatus,
+      );
+      designationFormModel.dynamicData =
+        designationAgreement.value.submittedData;
+      designationFormModel.locations = designationAgreement.value.locationsDesignations.map(
+        location => ({
+          locationId: location.locationId,
+          locationName: location.locationName,
+          requestForDesignation: location.requested,
+          approvedForDesignation: location.approved,
+          locationAddress: formatter.getFormattedAddress({
+            ...location.locationData.address,
+            provinceState: location.locationData.address.province,
+          }),
+        }),
+      );
+    };
+    onMounted(async () => {
+      await loadDesignation();
+    });
+
+    const updateDesignation = async (
+      designationStatus: DesignationAgreementStatus,
+    ) => {
+      if (designationStatus === DesignationAgreementStatus.Approved) {
+        /*If the update action is Approval, the build the designation location array for form
+          by merging the institution locations list with designation locations list
+        */
+        const institutionLocations = await InstitutionService.shared.getAllInstitutionLocationSummary(
+          designationAgreement.value.institutionId,
+        );
+        updateDesignationModel.value.startDate =
+          designationAgreement.value.startDate;
+        updateDesignationModel.value.endDate =
+          designationAgreement.value.endDate;
+        updateDesignationModel.value.designationStatus = designationStatus;
+        updateDesignationModel.value.institutionId =
+          designationAgreement.value.institutionId;
+        updateDesignationModel.value.locationsDesignations = institutionLocations?.map(
+          location =>
+            ({
+              locationId: location.id,
+              locationName: location.name,
+              requested: false,
+              locationAddress: formatter.getFormattedAddress({
+                ...location.data.address,
+                provinceState: location.data.address.province,
+              }),
+            } as UpdateDesignationLocationDto),
+        );
+        updateDesignationModel.value.locationsDesignations.forEach(location => {
```

Comment:
Can this code be simplified with something like below
![image](https://user-images.githubusercontent.com/61259237/156669966-24771154-e0c5-4c70-a356-2a76f7ac7c61.png)

Reply:
Logic adjusted as discussed.